http.go

```diff
@@ -6,67 +6,327 @@ package revel
+}
+
+func (req *Request) GetRequestURI() string {
+	uri, _ := req.GetValue(HTTP_REQUEST_URI).(string)
+	return uri
+}
+func (req *Request) GetQuery() (v url.Values) {
+	v, _ = req.GetValue(ENGINE_PARAMETERS).(url.Values)
+	return
+}
+func (req *Request) GetPath() (path string) {
+	path, _ = req.GetValue(ENGINE_PATH).(string)
+	return
+}
+func (req *Request) GetBody() (body io.Reader) {
+	body, _ = req.GetValue(HTTP_BODY).(io.Reader)
+	return
+}
+
+// Deprecated use controller.Params.Get()
+func (req *Request) FormValue(key string) (value string) {
+	return req.controller.Params.Get(key)
+}
+
+// Deprecated use controller.Params.Form[Key]
+func (req *Request) PostFormValue(key string) (value string) {
+	valueList := req.controller.Params.Form[key]
+	if len(valueList)>0 {
+		value = valueList[0]
+	}
+	return 
+}
+
+// Deprecated use GetForm() instead
+func (req *Request) ParseForm() (e error) {
+	if req.Form==nil {
+		req.Form, e = req.GetForm()
 	}
+	return
+}
+
+func (req *Request) GetForm() (url.Values, error) {
+	if form, err := req.In.Get(HTTP_FORM); err != nil {
+		return nil, nil
+	} else if values, found := form.(url.Values); found {
+		req.Form = values
+		return values, nil
+	}
+	return nil, FORM_NOT_FOUND
+}
+
+// Deprecated for backwards compatibility only
+type MultipartForm struct {
+	File   map[string][]*multipart.FileHeader
+	Value  url.Values
+	origin ServerMultipartForm
+}
+
+func (r *Request) MultipartReader() (*multipart.Reader, error) {
+
+	return nil, errors.New("MultipartReader not supported, use controller.Param")
+}
+
+// Deprecated for backwards compatibility only
+func newMultipareForm(s ServerMultipartForm) (f *MultipartForm) {
+	return &MultipartForm{File: s.GetFile(), Value: s.GetValue(), origin: s}
+}
+
+// Deprecated use GetMultipartForm() instead
+func (req *Request) ParseMultipartForm(_ int64) (e error) {
+	var s ServerMultipartForm
+	if s, e = req.GetMultipartForm(); e == nil {
+		req.MultipartForm = newMultipareForm(s)
+	}
+	return
+}
+
+// Return the args for the context
+func (req *Request) Context() map[string]interface{} {
+	return req.controller.Args
+}
+
+func (req *Request) GetMultipartForm() (ServerMultipartForm, error) {
+	if form, err := req.In.Get(HTTP_MULTIPART_FORM); err != nil {
+		return nil, nil
+	} else if values, found := form.(ServerMultipartForm); found {
+		return values, nil
+	}
+	return nil, FORM_NOT_FOUND
+}
+func (req *Request) Destroy() {
+	req.In = nil
+	req.ContentType = ""
+	req.Format = ""
+	req.AcceptLanguages = nil
+	req.Method = ""
+	req.RemoteAddr = ""
+	req.Host = ""
+	req.Header.Destroy()
+	req.URL = nil
+	req.Form = nil
+	req.MultipartForm = nil
+}
+
+func (resp *Response) SetResponse(r ServerResponse) {
+	resp.Out.Server = r
+	if h, e := r.Get(HTTP_SERVER_HEADER); e == nil {
+		resp.Out.internalHeader.Server, _ = h.(ServerHeader)
+	}
+}
+func (o *OutResponse) Destroy() {
+	o.response = nil
+	o.internalHeader.Destroy()
+}
+func (h *RevelHeader) Destroy() {
```

Comment:
I am torn, generally I like the type definitions at the top of the file and the functions following them. Of course this approach has issues when multiple type definitions, what is your thoughts ?